File: travelAgency2/travelAgency2/Repository/UserRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using travelAgency2.Repository.Database;
using travelAgency2.src.Domain;
using travelAgency2.src.Repository;
using travelAgency2.Utils;

namespace travelAgency2.Repository
{
    internal class UserRepository : Repository<long, User>
    {
        private Table<long, User> table;

        public UserRepository()
        {
            table = (UserTable) Resources.getTableFactory().getTable(Constants.Db.Tables.USER);
         }
           
        public void add(User entity)
        {
            table.add(entity);
        }

        public void delete(long id)
        {
            throw new NotImplementedException();
        }

        public User findByID(long id)
        {
            return table.findById(id);
        }

        public List<User> getAll()
        {
            return table.getAll();
        }

        public void update(User entity)
        {
            throw new NotImplementedException();
        }
    }
}

File: travelAgency/src/main/java/group/travelagency/repository/UserRepository.java
package group.travelagency.repository;

import group.travelagency.domain.User;
import group.travelagency.repository.database.Table;
import group.travelagency.repository.database.UserTable;
import group.travelagency.utils.Constants;
import group.travelagency.utils.Resources;

import java.sql.SQLException;
import java.util.List;
import java.util.Optional;

public class UserRepository implements Repository<Long, User>{
    private final UserTable table;

    public UserRepository() throws SQLException {
        table = (UserTable) Resources.getTableFactory().getTable(Constants.Db.Tables.USER);
    }

    @Override
    public void add(User entity) {
        table.add(entity);
    }

    @Override
    public void update(User entity, User newEntity) {

    }


    @Override
    public Optional<User> findByID(Long id) {
        return table.findById(id);
    }

    @Override
    public void delete(Long id) {
        table.delete(id);
    }

    @Override
    public List<User> getAll() {
            return table.getAll();
    }

    public Table<Long, User> getTable() {
        return table;
    }

    public Optional<User> findUser(String username, String password) {
        return table.findUser(username, password);
    }
}
